dbms/src/TestUtils/MPPTaskTestUtils.h

```diff
@@ -14,50 +14,90 @@
 
 #pragma once
 
+#include <Debug/MockComputeServerManager.h>
+#include <Debug/MockStorage.h>
 #include <Server/FlashGrpcServerHolder.h>
 #include <TestUtils/ExecutorTestUtils.h>
 
 namespace DB::tests
 {
+class MockTimeStampGenerator : public ext::Singleton<MockTimeStampGenerator>
+{
+public:
+    Int64 nextTs()
+    {
+        return ++current_ts;
+    }
+
+private:
+    std::atomic<UInt64> current_ts = 0;
+};
+
+DAGProperties getDAGPropertiesForTest(int server_num)
+{
+    DAGProperties properties;
+    // enable mpp
+    properties.is_mpp_query = true;
+    properties.mpp_partition_num = server_num;
+    properties.start_ts = MockTimeStampGenerator::instance().nextTs();
+    return properties;
+}
 
 class MPPTaskTestUtils : public ExecutorTest
 {
 public:
     static void SetUpTestCase()
     {
         ExecutorTest::SetUpTestCase();
-        TiFlashSecurityConfig security_config;
-        TiFlashRaftConfig raft_config;
-        raft_config.flash_server_addr = "0.0.0.0:3930"; // TODO:: each FlashGrpcServer should have unique addr.
-        Poco::AutoPtr<Poco::Util::LayeredConfiguration> config = new Poco::Util::LayeredConfiguration;
         log_ptr = Logger::get("compute_test");
-        compute_server_ptr = std::make_unique<FlashGrpcServerHolder>(TiFlashTestEnv::getGlobalContext(), *config, security_config, raft_config, log_ptr);
     }
 
-    static void TearDownTestCase()
+    static void TearDownTestCase() // NOLINT(readability-identifier-naming))
+    {
+        MockComputeServerManager::instance().reset();
+    }
+
+    void startServers(std::vector<String> addrs)
+    {
+        MockComputeServerManager::instance().reset();
+        auto size = std::thread::hardware_concurrency();
```

Comment:
Can we use a fixed server num?
or specified by test_context?

Reply:
> Can we use a fixed server num? or specified by test_context?

I will provide a default startServers method which will only be called in SetupTestCase().
And a startServers(int server_num ) for users who want to specify the num of servers, in this method the addr will be generated automatically.